Source/DataSources/KmlDataSource.js

```diff
@@ -1947,20 +1945,52 @@ define([
         Tour : processUnsupported
     };
 
+    var promisifiedFeatureTypes = {
+        // already promisified
+        Document : processDocument,
+        // already promisified
+        Folder : processFolder,
+        Placemark : deferAndWrapInPromise(processPlacemark),
+        NetworkLink : deferAndWrapInPromise(processNetworkLink),
+        GroundOverlay : deferAndWrapInPromise(processGroundOverlay),
+        PhotoOverlay : deferAndWrapInPromise(processUnsupported),
+        ScreenOverlay : deferAndWrapInPromise(processUnsupported),
+        Tour : deferAndWrapInPromise(processUnsupported)
+    };
+
+    function deferAndWrapInPromise(originalFunction){
+        return function(){
+            var deferred = when.defer();
+            var args = Array.prototype.slice.call(arguments);
+            var context = this;
+            setTimeout(function deferredFunc(){
```

Comment:
@mramato I've been churning it over and I think our best bet is still to use `setTimeout`, at least for now.  Tying it too closely with the render loop would have unintended side effects (as @kring pointed out).  It will impact performance, but at the browsers discretion.  In my testing, the browser still runs the render loop (and other browser stuff like mouse clicks etc) while this processing is going on.